sdk/core/azure-core/test/ut/string_test.cpp

```diff
@@ -131,3 +131,112 @@ TEST(String, toUpper)
   EXPECT_NE(StringExtensions::ToUpper("a"), "aA");
   EXPECT_NE(StringExtensions::ToUpper("abc"), "abcd");
 }
+
+TEST(String, SplitEmpty)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split("", ',');
+    EXPECT_TRUE(result.empty());
+  }
+  {
+    std::vector<std::string> result = StringExtensions::Split("", ',', true);
+    EXPECT_TRUE(result.empty());
+  }
+}
+
+TEST(String, Split3)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,2,3", ',');
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,2,3", ',', true);
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+}
+
+TEST(String, SplitBegin)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split(",1,2,3", ',');
+    ASSERT_EQ(result.size(), size_t{4});
+    EXPECT_EQ(result[0], "");
+    EXPECT_EQ(result[1], "1");
+    EXPECT_EQ(result[2], "2");
+    EXPECT_EQ(result[3], "3");
+  }
+  {
+    std::vector<std::string> result = StringExtensions::Split(",1,2,3", ',', true);
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+}
+
+TEST(String, SplitEnd)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,2,3,", ',');
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,2,3,", ',', true);
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+}
+
+TEST(String, SplitPartEmpty)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,,2,,3", ',');
+    ASSERT_EQ(result.size(), size_t{5});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "");
+    EXPECT_EQ(result[2], "2");
+    EXPECT_EQ(result[3], "");
+    EXPECT_EQ(result[4], "3");
+  }
+  {
+    std::vector<std::string> result = StringExtensions::Split("1,,2,,3", ',', true);
+    ASSERT_EQ(result.size(), size_t{3});
+    EXPECT_EQ(result[0], "1");
+    EXPECT_EQ(result[1], "2");
+    EXPECT_EQ(result[2], "3");
+  }
+}
+
+TEST(String, SplitSeparator)
+{
+  using Azure::Core::_internal::StringExtensions;
+  {
+    std::vector<std::string> result = StringExtensions::Split(",,,,", ',');
+    ASSERT_EQ(result.size(), size_t{4});
+    EXPECT_EQ(result[0], "");
```

Comment:
You might want to add a case looking for something that looks like a connection string:

Foo=Bar;Bas=Bletch;Other=SEREWVEWD==

I got burned yesterday because I used Split on (';') and then Split on "=". Which did very bad things on the "Other" field.

Reply:
one could argue that it did exactly what was expected, even the .net one will remove the == at the end

Reply:
if you don't like it i'll remove it from core and we can continue to spread the split string through the code